src/FrequencyMapAgent.cpp

```diff
@@ -255,6 +284,13 @@ namespace geopm
             }
             m_platform_io.adjust(m_uncore_min_ctl_idx, m_uncore_init_min);
             m_platform_io.adjust(m_uncore_max_ctl_idx, m_uncore_init_max);
+
+            for (size_t ctl_idx = 0; ctl_idx < (size_t)m_num_gpu_freq_ctl_domain; ++ctl_idx) {
```

Comment:
Change this to an if() and push the GPU signals initially at the board scope in the init() method.